src/EditorFeatures/CSharp/BlockCommentEditing/CloseBlockCommentCommandHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.BlockCommentEditing
+{
+    [Export(typeof(VSCommanding.ICommandHandler))]
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [Name(nameof(CloseBlockCommentCommandHandler))]
+    [Order(After = nameof(BlockCommentEditingCommandHandler))]
+    internal sealed class CloseBlockCommentCommandHandler : VSCommanding.ICommandHandler<TypeCharCommandArgs>
+    {
+        public string DisplayName => EditorFeaturesResources.Block_Comment_Editing;
+
+        public bool ExecuteCommand(TypeCharCommandArgs args, VSCommanding.CommandExecutionContext executionContext)
+        {
+            if (args.TypedChar == '/')
+            {
+                var caret = args.TextView.GetCaretPoint(args.SubjectBuffer);
+                if (caret != null)
+                {
+                    var (snapshot, position) = caret.Value;
+
+                    if (position >= 2 &&
```

Comment:
Suggestion: this logic, and the nested logic that checks to see if the line before the '*' is whitespace could use a comment to describe the behavior.

Reply:
➡️ Comment added in 42baa98e7fb23c6820fa7dccb247b0fedb7c6374